icu4c/source/common/dictbe.cpp

```diff
@@ -1077,14 +1078,20 @@ CjkBreakEngine::CjkBreakEngine(DictionaryMatcher *adoptDictionary, LanguageType
         UnicodeSet cjSet(UnicodeString(u"[[:Han:][:Hiragana:][:Katakana:]\\u30fc\\uff70\\uff9e\\uff9f]"), status);
         if (U_SUCCESS(status)) {
             setCharacters(cjSet);
+
+#if !UCONFIG_NO_CJ_ML_PHRASE_BREAKING
+            fMlBreakEngine = new CjMLBreakEngine(fDigitOrOpenPunctuationOrAlphabetSet, fClosePunctuationSet, status);
```

Comment:
you need to check if fMlBreakEngine is nullptr after the new and set status to U_MEMORY_ALLOCATION_ERROR  if it is null and return . Otherwise, your later fMlBreakEngine->divideUpDictionaryRange() would crash

Reply:
Done.